src/Workspaces/Core/Portable/Shared/Utilities/BloomFilter.cs

```diff
@@ -395,4 +398,73 @@ private static bool IsEquivalent(BitArray array1, BitArray array2)
 
         return true;
     }
+
+    /// <summary>
+    /// Provides mechanism to efficiently obtain bloom filter hash for a value. Backed by a single element cache.
+    /// </summary>
+    internal class BloomFilterHash
+    {
+        private static BloomFilterHash? s_cachedHash;
+
+        private readonly string _value;
+        private readonly bool _isCaseSensitive;
+        private readonly ImmutableArray<int> _hashes;
+
+        private BloomFilterHash(string value, BloomFilter filter)
+        {
+            _value = value;
+            _isCaseSensitive = filter._isCaseSensitive;
+
+            var hashBuilder = new FixedSizeArrayBuilder<int>(filter._hashFunctionCount);
+
+            for (var i = 0; i < filter._hashFunctionCount; i++)
+                hashBuilder.Add(filter.GetBitArrayIndex(value, i));
+
```

Comment:
This seems wrong (though it is early). The result of GetBitArrayIndex depends on internal state of the filter it doesn't seem like it could be used across filters.